Chap_API_Sets_Groups.tex

```diff
@@ -4,38 +4,99 @@
+    \ac{PMIx} servers; and
+    \item that process set names do not conflict with system-assigned namespaces within the scope of the set
+\end{itemize}
+
+\advicermend
+
+Process \emph{sets} differ from process \emph{groups} in several key ways:
 
 \begin{itemize}
     \item Process \emph{sets} have no implied relationship between their members - i.e., a process in a process set has no concept of a ``pset rank'' as it would in a process \emph{group}
-    \item Process \emph{set} identifiers are considered job-level information set at launch. No \ac{PMIx} \ac{API} is provided by which a user can change the process \emph{set} value of a process on-the-fly. In contrast, \ac{PMIx} process \emph{groups} can only be defined dynamically by the application.
+
+    \item Process \emph{set} identifiers are set by the host environment -
+    there are no \ac{PMIx} \acp{API} provided by which an application can
+    change a process \emph{set} membership. In contrast, \ac{PMIx} process
+    \emph{groups} can only be defined dynamically by the application.
+
     \item Process \emph{groups} can be used in calls to \ac{PMIx} operations. Members of process \emph{groups} that are involved in an operation are translated by their \ac{PMIx} server into their \emph{native} identifier prior to the operation being passed to the host environment. For example, an application can define a process group to consist of ranks 0 and 1 from the host-assigned namespace of \emph{210456}, identified by the group id of \emph{foo}. If the application subsequently calls the \refapi{PMIx_Fence} \ac{API} with a process identifier of \{foo, PMIX_RANK_WILDCARD\}, the \ac{PMIx} server will replace that identifier with an array consisting of \{210456, 0\} and \{210456, 1\} - the host-assigned identifiers of the participating processes - prior to passing the request up to the host environment
+
     \item Process \emph{groups} can request that the host environment assign a unique \code{size_t} \ac{PGCID} to the group at time of group construction. An \ac{MPI} library may, for example, use the \ac{PGCID} as the \ac{MPI} communicator identifier for the group.
 \end{itemize}
 
-The two concepts do, however, overlap in one specific area. Process \emph{groups} are included in the process \emph{set} information returned by calls to \refapi{PMIx_Query_info_nb}. Thus, a \emph{process group} can effectively be considered an extended version of a \emph{process set} that adds dynamic definition and operational context to the \emph{process set} concept.
+The two concepts do, however, overlap in one specific area. Process
+\emph{groups} are included in the process \emph{set} information returned by
+calls to \refapi{PMIx_Query_info_nb} with the \refattr{PMIX_PSET_NAMES}.
+Thus, a \emph{process group} can effectively be considered an extended version
+of a \emph{process set} that adds dynamic definition and operational context
+to the \emph{process set} concept.
 
```

Comment:
I'm not sure I like combining both progress set and progress group identifiers in the set returned by `PMIX_PSET_NAMES` because i don't know how the user would sort out which one is which. In particular if another client-side library defined a progress group then another client-side library would not be able to tell if the string returned was a process set (defined by the host) or a progress group (defined by another part of the app).

Alternatively, can we create a `PMIX_PGROUP_NAMES` that returns the process groups, and then leave `PMIX_PSET_NAMES` as just the process set names?

Reply:
Done in latest commit